frontend/src/Context/PublisherProvider/usePublisherOptions/usePublisherOptions.spec.tsx

```diff
@@ -42,57 +44,90 @@ const mockUserContextWithCustomSettings = {
 } as UserContextType;
 
 describe('usePublisherOptions', () => {
+  let enumerateDevicesMock: ReturnType<typeof vi.fn>;
+  let deviceManager: DeviceManager;
+  beforeEach(async () => {
+    enumerateDevicesMock = vi.fn();
+    vi.stubGlobal('navigator', {
+      mediaDevices: {
+        enumerateDevices: enumerateDevicesMock,
+      },
+    });
+    Object.defineProperty(window, 'localStorage', {
+      value: localStorageMock,
+      writable: true,
+    });
+    deviceManager = new DeviceManager();
+    enumerateDevicesMock.mockResolvedValue([]);
+    await deviceManager.init();
+  });
+
+  afterAll(() => {
+    window.localStorage.clear();
+  });
+
   it('should use default settings', async () => {
     vi.spyOn(OT, 'hasMediaProcessorSupport').mockReturnValue(true);
     mockUseUserContext.mockImplementation(() => mockUserContextWithDefaultSettings);
     const { result } = renderHook(() => usePublisherOptions());
-    expect(result.current).toEqual({
-      resolution: '1280x720',
-      publishAudio: false,
-      publishVideo: false,
-      audioSource: undefined,
-      videoSource: undefined,
-      insertDefaultUI: false,
-      audioFallback: {
-        publisher: true,
-      },
-      audioFilter: {
-        type: 'advancedNoiseSuppression',
-      },
-      videoFilter: undefined,
-      name: '',
-      initials: '',
+    await waitFor(() => {
+      expect(result.current).toEqual({
+        resolution: '1280x720',
+        publishAudio: false,
+        publishVideo: false,
+        audioSource: undefined,
+        videoSource: undefined,
+        insertDefaultUI: false,
+        audioFallback: {
+          publisher: true,
+        },
+        audioFilter: {
+          type: 'advancedNoiseSuppression',
+        },
+        videoFilter: undefined,
+        name: '',
+        initials: '',
+      });
     });
   });
 
   it('should not have advanced noise suppression if not supported by browser', async () => {
     vi.spyOn(OT, 'hasMediaProcessorSupport').mockReturnValue(false);
     mockUseUserContext.mockImplementation(() => mockUserContextWithDefaultSettings);
     const { result } = renderHook(() => usePublisherOptions());
-    expect(result.current.audioFilter).toBe(undefined);
+    expect(result.current?.audioFilter).toBe(undefined);
```

Comment:
since I changed the `usePublisherOptions` to be potentially `null` - in order to catch the case that the audio source and video source are not set, this needed a `?`

Reply:
Does it need to be nullable?  Isn't an empty object sufficient?

Reply:
I think it does in this case because of using this `useEffect` [here](https://github.com/Vonage/vonage-video-react-app/pull/139/files#diff-2bf3c35c2aa956b3ba82d7b5bea5842607a6d703df923b5ecc4ec958b427d261R101-R108) and [here](https://github.com/Vonage/vonage-video-react-app/pull/139/files#diff-ed35da1d82a0249b96617d09ffc6eb1e33f80541beed2e7292d0371466b2fac6R22-R27).

if I do `publisherOptions` as an empty object instead of null doing `if (!publisherOptions)` won't work and doing stuff like `if (Object.keys(publisherOptions).length === 0)` isn't pretty